src/gui/components.py

```diff
@@ -26,23 +29,98 @@ def add_qrcode(text, y=QR_PADDING, scr=None, style=None, width=None):
     qr = QRCode(scr)
     qr.set_text(text)
     qr.set_size(width)
-    qr.set_text(text)
+    qr.set_text(None)
     qr.align(scr, lv.ALIGN.IN_TOP_MID, 0, y)
+
+    if qr.buttons != None:
+        # FIXME dont know where the QR code is
+        assert qr.indx_msg != None
+        BTN_PADDING = 30
+        qr.buttons[0].align(scr, lv.ALIGN.IN_TOP_MID, -(width//2 + BTN_PADDING), width//2)
+        qr.buttons[1].align(scr, lv.ALIGN.IN_TOP_MID, (width//2 + BTN_PADDING), width//2)
+        qr.indx_msg.align(qr, lv.ALIGN.IN_TOP_MID, 0, 20)
```

Comment:
i will have to throw this out as i dont know where the qr is to align it with the buttons. The buttons can only be aligned in QRAlert(). So add_qrcode() will always have to be called in QRAlert. Yes?